packages/core/src/render3/instructions/defer.ts

```diff
@@ -605,3 +629,79 @@ export interface DeferBlockDependencyInterceptor {
 export const DEFER_BLOCK_DEPENDENCY_INTERCEPTOR =
     new InjectionToken<DeferBlockDependencyInterceptor>(
         ngDevMode ? 'DEFER_BLOCK_DEPENDENCY_INTERCEPTOR' : '');
+
+/**
+ * Determines if a given value matches the expected structure of a defer block
+ *
+ * We can safely rely on the primaryTmplIndex because every defer block requires
+ * that a primary template exists. All the other template options are optional.
+ *
+ * @param value
+ * @returns boolean
+ */
+function isTDeferBlockDetails(value: unknown): value is TDeferBlockDetails {
+  return (typeof value === 'object') &&
+      (typeof (value as TDeferBlockDetails).primaryTmplIndex === 'number');
+}
+
+/**
+ * Configuration of defer block behavior in TestBed.
+ */
+export interface DeferBlockConfig {
+  initialState: string;
+}
+
+/**
+ * Token used for configuring test bed defer block behavior.
+ */
+export const DEFER_BLOCK_CONFIG =
+    new InjectionToken<DeferBlockConfig>(ngDevMode ? 'DEFER_BLOCK_CONFIG' : '');
+
+/**
+ * Defer block instance for testing.
+ */
+export interface DeferBlockDetails {
+  lContainer: LContainer;
+  lView: LView;
+  tNode: TNode;
+  details: TDeferBlockDetails;
```

Comment:
For consistency with other fields, we cam include `t` in front of `details`:
```suggestion
  tDetails: TDeferBlockDetails;
```